### ISSUE TITLE
自定义损失函数无法自动求梯度。

### ISSUE THREAD

**Issue by scott0o0** [PaddlePaddle/Paddle#26378]
为使您的问题得到快速解决，在建立Issues前，请您先通过如下方式搜索是否有相似问题:【搜索issue关键字】【使用labels筛选】【官方文档】

如果您没有查询到相似问题，为快速解决您的提问，建立issue时请提供如下细节信息：
- 标题：简洁、精准概括您的问题，例如“Insufficient Memory xxx" ”
- 版本、环境信息：
    1）PaddlePaddle版本：paddlepaddle 1.8
    3）GPU：2080ti   cuda10.0
    4）系统环境：请您描述系统类型、版本，例如Mac OS 10.14，Python版本

- 问题描述：

    想对yolov3计算的损失函数loss = fluid.layers.yolov3_loss()进行补充，增加一项损失与原先的损失之和作为总损失函数total_loss = loss + loss_add。
    其中loss_add中有一项是需要计算两个不同层级输出的图像特征矩阵的2范数：||X-Y||^2（两个图像特征的shape一致，如下图所示），可我无法做到计算完成之后保留梯度特征（stop_gradient=False 这一行就不见了）
![image](https://user-images.githubusercontent.com/49825779/90401260-45609b80-e0d0-11ea-8e6a-07ecd4101eea.png)
在官方文档中也找不到自定义损失函数的示例。希望大佬可以提供一下解决方法，可以使修改后的损失函数能够自动求梯度，不胜感激!

**Comment by scott0o0** [comment]
补充一下：计算||X-Y||的目的是计算两个矩阵的欧式距离，即假设矩阵T的大小为[M,D]，训练集矩阵P的大小为[N,D]（测试集中共有M个点，每个点为D维特征向量；训练集中共有N个点，每个点为D维特征向量），计算距离dists。
![image](https://user-images.githubusercontent.com/49825779/90511958-e4000180-e18f-11ea-91ab-8853041df1f8.png)

所需的一些其他函数如指数运算可以用paddle.fluid.layers.exp(x, name=None)，试了一下并不会使梯度的记录消失。觉得应该可以直接使用吧。^_^
